aaelf32/aaelf32.rst

```diff
@@ -0,0 +1,3289 @@
+  [ARMARM_].
+
+Relocation types
+~~~~~~~~~~~~~~~~
+
+The following table lists the relocation codes for Arm. The table shows:
+
+* The code which is stored in the ELF32_R_TYPE component of the r_info field.
+
+* The mnemonic name for the relocation.
+
+* The type of the relocation.  This field substantially divides the relocations
+  into Static and Dynamic relocations.  Static relocations are processed by a
+  static linker; they are normally either fully resolved or used to produce
+  dynamic relocations for processing by a post-linking step or a dynamic
+  loader. A well formed image will have no static relocations after static
+  linking is complete, so a post-linker or dynamic loader will normally only
+  have to deal with dynamic relocations.  This field is also used to describe
+  deprecated, obsolete, private and unallocated relocation codes.  Deprecated
+  codes should not be generated by fully conforming toolchains; however it is
+  recognized that there may be substantial existing code that makes use of
+  these forms, so it is expected that a linker may well be required to handle
+  them at this time.  Obsolete codes should not be used, and it is believed
+  that there is little or no common use of these values.  All unallocated codes
+  are reserved for future allocation.
+
+* The class of the relocation describes the type of place being relocated:
+  these are Data, Arm, Thumb16 and Thumb32 (32-bit long-format instructions).
+  A special class of Miscellaneous is used when the operation is not a simple
+  mathematical expression.
+
+* The operation field describes how the symbol and addend are processed by the
+  relocation code.  It does not describe how the addend is formed (for a REL
+  type relocation), what overflow checking is done, or how the value is written
+  back into the place: this information is given in subsequent sections. In all
+  cases, relocation expression values are computed mod 2\ :sup:`32`.
+
+The following nomenclature is used for the operation:
+
+* S (when used on its own) is the address of the symbol.
+
+* A is the addend for the relocation.
+
+* P is the address of the place being relocated (derived from r_offset).
+
+* Pa is the adjusted address of the place being relocated, defined as (P &
+  0xFFFFFFFC).
+
+* T is 1 if the target symbol S has type STT_FUNC and the symbol addresses a
+  Thumb instruction; it is 0 otherwise.
+
+* B(S) is the addressing origin of the output segment defining the symbol S.
+  The origin is not required to be the base address of the segment. This value
+  must always be word-aligned.
+
+* GOT_ORG is the addressing origin of the Global Offset Table (the indirection
+  table for imported data addresses). This value must always be word-aligned.
+  See `Proxy generating relocations`_.
+
+* GOT(S) is the address of the GOT entry for the symbol S.
+
+.. _Relocation codes table:
+
+.. table:: Relocation codes
+  :class: longtable
```

Comment:
as described above on the version history table, please use a class to specify the width, and longtable doesn't exist here (at least we're not using it if it might be defined).